---
title: "cicero-addition"
knit: (function(input_file, encoding) {
  out_dir <- 'docs';
  rmarkdown::render(input_file,
 encoding=encoding,
 output_file=file.path(dirname(input_file), out_dir, 'index.html'))})
output: html_document
---

##Setup Library
```{r eval=false}
knitr::opts_knit$set(root.dir = normalizePath("D:/R/peakAnalysisFresh")) 
```

##Set Constants For Peak Selection
```{r eval=false}
thresh = 5
changePerc = 40
```

##Neccesary Libraries
```{r eval=false}
devtools::install_github("cole-trapnell-lab/L1-graph")
install.packages("BiocManager")
BiocManager::install("monocle", version = "3.8")
library("monocle")
library("cicero")
library("dplyr")
library("tidyr")
library("lattice")
library("raster")
```

##Install and Setup Cicero Additions Package
```{r eval=false}
devtools::install_github("samissimas/ciceroAddition")
library(ciceroAdditions)
```

##Load CDS and Generate Chromosone File
```{r eval=false}
load("WT_cicero_cds.rda")
load("KO_cicero_cds.rda")

mm10.chr <- read.delim("C:/Users/samro/Desktop/BAM Files/Cicero/chromosome_size.txt", header=F, sep="\t")
```

##Conns Generation Loop
```{r eval=false}
counter = 1
while(counter <= 19)
{
  
charcount = as.character(counter)
j = paste("chr",charcount, sep = "")
sample_genome <- subset(mm10.chr, V1==j)
WT_conns <- run_cicero(WT_cicero_cds, sample_genome)
KO_conns <- run_cicero(KO_cicero_cds, sample_genome)
KOchar = paste("KO_conns_",j,".rda", sep = "")
WTchar = paste("WT_conns_",j,".rda", sep = "")
save(KO_conns, file = KOchar)
save(WT_conns, file = WTchar)
counter = counter + 1

}
```

##Primary Analysis Loop (Create The Peak Blends Files For Each Chromosone) [Most Time Intensive] 
```{r eval=false}
counter = 19
while(counter <= 19)
{
  
charcount = as.character(counter)
printer = paste("Beginning:", as.character(counter), sep = " ")
print(printer)
j = paste("chr",charcount, sep = "")
KOchar = paste("KO_conns_",j,".rda", sep = "")
WTchar = paste("WT_conns_",j,".rda", sep = "")
load(KOchar)
load(WTchar)

print("loaded connections")

uniquePeaksKO = unique_peak(KO_conns) 

print("unique peaks for KO done")

uniquePeaksWT = unique_peak(WT_conns)

print("unique peaks for WT done")

vectorsizeKO <- length(uniquePeaksKO)
vectorsizeWT <- length(uniquePeaksWT)

KOpeaks.data <- createDF(vectorsizeKO)
WTpeaks.data <- createDF(vectorsizeWT)

KOpeaks.data <- buildDF(uniquePeaksKO, KOpeaks.data)
WTpeaks.data <- buildDF(uniquePeaksWT, WTpeaks.data)

SUMpeaks.data = sumPeaksName(WTpeaks.data, KOpeaks.data, vectorsizeKO, "W","K")

print("sum peaks done")

SUMpeaksblend.data = peakBlend(SUMpeaks.data)

SUMcharblend = paste(j,"_","SUMpeaks_Blend", ".rda", sep = "")
save(SUMpeaksblend.data, file = SUMcharblend)
counter = counter + 1
}
```

##Secondary Analysis Loop (Do All Connection Weighting) [Second Most Time Intensive]
```{r eval=false}
counter = 1
while(counter <= 1)
{

charcount = as.character(counter)
j = paste("chr",charcount, sep = "")

KOchar = paste("KO_conns_",j,".rda", sep = "")
WTchar = paste("WT_conns_",j,".rda", sep = "")
load(KOchar)
load(WTchar)

print("connections loaded")

SUMcharblend = paste(j,"_","SUMpeaks_Blend", ".rda", sep = "")
load(SUMcharblend)

print("blend peaks loaded")

SUMpeaksKO.data <- connblendInt(SUMpeaksblend.data, KO_conns, .2)

print("KO connections tabulated")

SUMpeaksWT.data <- connblendInt(SUMpeaksblend.data, WT_conns, .2)

print("WT connections tabulated")

colnames(SUMpeaksKO.data)[colnames(SUMpeaksKO.data)=="access"] <- "KO_access"
colnames(SUMpeaksWT.data)[colnames(SUMpeaksWT.data)=="access"] <- "WT_access"
colnames(SUMpeaksKO.data)[colnames(SUMpeaksKO.data)=="connections"] <- "KO_connections"
colnames(SUMpeaksWT.data)[colnames(SUMpeaksWT.data)=="connections"] <- "WT_connections"


SUMpeaksPreSort.data <- cbind(SUMpeaksKO.data, SUMpeaksWT.data[!names(SUMpeaksKO.data) %in% names(SUMpeaksWT.data)])

SUMpeaksPreSort.data = SUMpeaksPreSort.data[,c(1,2,3,4,5,7,6,8)]

SUMcharpre = paste(j,"_","SUMpeaks_PreSort", ".rda", sep = "")
save(SUMpeaksPreSort.data, file = SUMcharpre)

print("beginning to find sig peaks")

SUMpeaksSorted.data <- sigPeakFinder(SUMpeaksPreSort.data, thresh, changePerc)

print("sum peaks weighted done")

SUMcharpost = paste(j,"_","SUMpeaks_thresh", as.character(thresh),"_change", as.character(changePerc), ".rda", sep = "")
save(SUMpeaksSorted.data, file = SUMcharpost)

counter = counter+1
}

```

##Loop To Combine Sorted Peak Data
```{r eval=false}
counter = 1
fullGenomeSorted <- data.frame()

while(counter <= 19)
{

charcount = as.character(counter)
j = paste("chr",charcount, sep = "")
  
SUMcharpost = paste(j,"_","SUMpeaks_thresh", as.character(thresh),"_change", as.character(changePerc), ".rda", sep = "")
load(file = SUMcharpost)
printer = paste("Loaded Data Frame:", as.character(counter), sep = "")
print(printer)
if(counter==1)
{
  fullGenomeSorted = SUMpeaksSorted.data
  
}
else{
  fullGenomeSorted  = rbind(fullGenomeSorted, SUMpeaksSorted.data)
}

counter = counter + 1
  
}
```

##Print Final Sorted Genome Result
```{r eval=false}
print(fullGenomeSorted)
```

##Genome Sorter (Further Sort Your Final Genome Result Based on A Bed File Of Interested Regions)
```{r eval=false}
bed <- as.data.frame(read.table("E13_WT1_c1_7_9_peaks_with_p63ChIP_with_p63motif_loss_less_than_2fold_in_KO.bed",header = FALSE, sep="\t",stringsAsFactors=FALSE, quote=""))

bed.data = buildDF_bed(bed)

i=1
j=1
length = dim(fullGenomeSorted)[1]
lengthy = dim(bed.data)[1]

newFullGenomeSorted <- data.frame()

while(i <= length)
{
  cicString2 = fullGenomeSorted[i,1]
    chromnum2 <- regexpr("r",cicString2)
    number2 <- substring(cicString2, chromnum2+1)
    
  if(as.numeric(number2) == bed.data[j,1])
  {
      if(as.numeric(fullGenomeSorted[i,2]) <= as.numeric(bed.data[j,3]))
      {
        if(as.numeric(fullGenomeSorted[i,3]) >= as.numeric(bed.data[j,2]))
        {
          newFullGenomeSorted = rbind(newFullGenomeSorted, fullGenomeSorted[i,])
          i = i + 1
        }
        else if(as.numeric(fullGenomeSorted[i,3]) < as.numeric(bed.data[j,2]))
        {
          i = i + 1
        }
      }
      else
      {
        j = j + 1
      }
  }
  else
  {
    if(as.numeric(number2) > bed.data[j,1])
    {
      j = j + 1
    }
    else if(as.numeric(number2) < bed.data[j,1])
    {
      i = i + 1
    }
  }
}
```

##Split the String Of Connections back into A Readable Data Object
```{r eval=false}

###Select which object you'd like to inspect
k=7
checkpeaks.data = list_split(SUMpeaksSorted.data[k,8])


```

##Graph A Reduced Dimension Version of Genome (Unique Peak Setup)
[Requires a Conn File Which you can generate here]
```{r eval=false}

chrom = 1

charcount = as.character(chrom)
j = paste("chr",charcount, sep = "")
KOchar = paste("KO_conns_",j,".rda", sep = "")
load(KOchar)



```

##Graph A Reduced Dimension Version of Genome (Matrix Setup)
```{r eval=false}
regionStart = 3611166
regionEnd = 185643524
reductionDim = 70

newPeaks <- simpleMatrix(KO_conns,reductionDim, .3, regionStart,regionEnd)
```

##Graph A Reduced Dimension Version of Genome
```{r eval=false}
graphTitle=paste("Connections From", as.character(regionStart), "to", as.character(regionEnd), sep = " ")
graphSub=paste("Pixels Represent Peaks Reduced in Dimension By:", as.character(reductionDim), sep = " ")


levelplot(newPeaks,col.regions = heat.colors(100)[length(heat.colors(100)):1], scales=list(x=list(at=NULL)), xlab="", ylab = "", main = graphTitle, sub = graphSub)

```

##Copmare Two Conns Files
```{r eval=false}
load("KO_conns_chr1.rda")
load("WT_conns_chr1.rda")


result = compareConns(WT_conns,KO_conns)

```



